dirty_equals/_other.py

```diff
@@ -369,3 +371,125 @@ def equals(self, other: Any) -> bool:
                 return False
 
         return True
+
+
+class IsEnum(DirtyEquals[Enum]):
+    """
+    Checks if an instance is an Enum.
+
+    Inherits from [`DirtyEquals`][dirty_equals.DirtyEquals].
+
+    ```py title="IsEnum"
+    from enum import Enum, auto
+    from dirty_equals import IsEnum
+
+    class ExampleEnum(Enum):
+        a = auto()
+        b = auto()
+
+    a = ExampleEnum.a
+
+    assert a == IsEnum
+    assert a == IsEnum(ExampleEnum)
+    ```
+    """
+
+    def __init__(self, enum_cls: Type[Enum] = Enum):
+        """
+        Args:
+            enum_cls: Enum class to check against.
+        """
+        self._enum_cls = enum_cls
+
+    def equals(self, other: Any) -> bool:
+        return isinstance(other, self._enum_cls)
```

Comment:
One idea to account for checking if `other` is a value in the `_enum_cls` could be:

```python
if isinstance(other, Enum):
    return isinstance(other, self._enum_cls)
else:
    return other in {i.value for i in self._enum_cls}
```